### ISSUE TITLE
"Next" type removed in version 0.5.1

### ISSUE THREAD

**Issue by DaniGuardiola** [lukesmurray/use-slate-with-extensions#13]
I relied on that type for a workaround:

```ts
import type {
  Next,
  OnKeyDown,
} from "use-slate-with-extensions";

const onKeyDown: OnKeyDown = <T,>(
  event: React.KeyboardEvent<T>,
  editor: SlateEditor,
  next?: Next<OnKeyDown>
) => {
  // ...
}
```

I believe this is because of an issue that was caused by the `OnKeyDown` type having `KeyboardEvent<{}>`. I did this to turn it into a generic which did fix my issue. I wonder if that should be part of the lib typings, or is there any particular reason the `{}` type is being used here instead of a generic?

**Comment by lukesmurray** [comment]
Fixed in 0.5.2

**Comment by lukesmurray** [comment]
Also made onKeyDown a generic.
